lib/util/http/queue.ts

```diff
@@ -0,0 +1,36 @@
+import URL from 'url';
+import PQueue from 'p-queue';
+import { getRequestLimit } from './host-rules';
+
+const hostQueues = new Map<string | null, PQueue | null>();
+
+function getUrlHost(url: string): string | null {
+  try {
+    return URL.parse(url).host;
+  } catch (e) {
+    return null;
+  }
+}
+
+export function getQueue(url: string): PQueue | null {
+  const host = getUrlHost(url);
+  if (!host) {
+    return null;
+  }
+
+  let queue = hostQueues.get(host);
+  if (queue === undefined) {
+    queue = null;
+    const concurrency = getRequestLimit(url);
+    if (concurrency) {
+      queue = new PQueue({ concurrency });
+    }
+  }
+  hostQueues.set(host, queue);
+
+  return queue;
+}
+
+export function clear(): void {
```

Comment:
This should be called before every repository, right?

Reply:
Added before `try-catch` block in `renovateRepo()`